job-server/test/spark.jobserver/routes/DataRoutesSpec.scala

```diff
@@ -39,8 +39,7 @@ class DataRoutesSpec extends WebApiSpec {
     }
 
     it("DELETE - should be able to remove file from tmp dir") {
-      val encodedName = URLEncoder.encode("/tmp/fileToRemove", "UTF-8")
-      Delete("/data/" + encodedName) ~> sealRoute(routes) ~> check {
+      Delete("/data//tmp/fileToRemove") ~> sealRoute(routes) ~> check {

```

Comment:
I think you still need to encode the filename here. The handler in WebApiSpec must be changed instead.
Simplest solution would be this:
instead of 
`    case DataManagerActor.StoreData("/tmp/fileToRemove", _) => sender ! DataManagerActor.Stored("/tmp/fileToRemove-time-stamp")`
`      case DataManagerActor.StoreData("errorfileToRemove", _) => sender ! DataManagerActor.Error`

it should be 
`    case DataManagerActor.StoreData("errorfileToRemove", _) => sender ! DataManagerActor.Error`     
`     case DataManagerActor.StoreData(filename, _) => sender ! DataManagerActor.Stored(filename + "-time-stamp")`

and similiarly for the DELETE cases.